src/watch/index.ts

```diff
@@ -83,36 +81,35 @@ export class Watcher extends EventEmitter {
 			.then(() => {
 				this.running = false;
 
-				if (this.dirty) {
+				if (this.rerun) {
+					this.rerun = false;
 					this.run();
```

Comment:
Maybe we should still add the usual pre-build timeout here before triggering another run in case there is ongoing file system activity. I.e. `invalidate` instead of `run`.

Reply:
Good catch, I've ensured builds never run faster than in intervals of DELAY.